app/src/main/java/com/getsensibill/spendmanager/demo/fragment/FragmentJavaActivity.java

```diff
@@ -31,54 +38,43 @@ protected void onCreate(Bundle savedInstanceState) {
         binding = ActivityFragmentJavaBinding.inflate(getLayoutInflater());
         setContentView(binding.getRoot());
 
-        if(savedInstanceState == null) {
-            WebUiFragment fragment = new WebUiFragment();
-            Bundle bundle = new Bundle();
-
-            // Pass in a navigation override. Defaults as .DASHBOARD
-            bundle.putParcelable(
-                    WebUiFragment.ARG_NAVIGATION_OVERRIDE,
-                    NavigationIntent.DASHBOARD.INSTANCE
-            );
-
-            // Pass in a custom Theme override. Defaults to null
-            WebTheme theme = new WebTheme(new Brand());
-            bundle.putParcelable(WebUiFragment.ARG_WEB_THEME_OVERRIDE, theme);
-
-            fragment.setArguments(bundle);
-
-            getSupportFragmentManager()
-                    .beginTransaction()
-                    .replace(binding.fragmentContainer.getId(), fragment, "FRAGMENT_TAG")
-                    .commit();
+        if (savedInstanceState == null) {
+            loadWebUi();
         }
     }
 
     /**
      * Required Override when using [WebUiFragment.Listener]
-     *
+     * <p>
      * Handle when the network error fragment needs to be navigated to. We have provided a
      * [WebUiNetworkErrorFragment] that you can display to the user. Or you can use your
      * own fragment or even an alternative logic flow if desired.
      */
     @Override
     public void onDisplayNetworkError(boolean networkNotAvailable) {
+        // reuses the fragment if it already existed
+
+        if (getFragment(TAG_WEB_NETWORK_ERROR_FRAGMENT) != null) {
```

Comment:
If you're reusing the error fragment here, you don't actually set in new args before showing the fragment on screen.
This means that if in one session, the error fragment is displayed multiple times (possibly for different reasons) - only the first configuration will ever be used (or maybe on the second time it may be completely unconfigured?  I'm not sure if android keeps `args` after detaching a fragment)

I suggest doing something like this instead:
```java
// If the fragment manager already has reference to an error fragment, reuse that instance.
// Otherwise, create a fresh one.
Fragment existingFragment = getFragment(TAG_WEB_NETWORK_ERROR_FRAGMENT);
WebUiNetworkErrorFragment fragment = existingFragment instanceof WebUiNetworkErrorFragment ?
        (WebUiNetworkErrorFragment) existingFragment :
        new WebUiNetworkErrorFragment();

Bundle args = new Bundle();

// If not provided, the fragment assumes that there is no network connection (default: true)
args.putBoolean(WebUiNetworkErrorFragment.ARG_NO_NETWORK_ERROR, networkNotAvailable);
fragment.setArguments(args);

setFragment(fragment, TAG_WEB_NETWORK_ERROR_FRAGMENT);
```

Reply:
updated